# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/GLImages.h
# Commit: 290917deea87
# Full Hash: 290917deea879f965f6f1ad441df945c41c01c14
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-12-13 21:42:49
# Regressor Bug: 1866020
# File Overlap Count: 1
# Description:
#   Bug 1866020 - Override buggy colorspace conversion on Pixel devices. r=gw,padenot,geckoview-reviewers,owlish
#   
#   Pixel 6, 7, and 8 devices running Android 14 are affected by a bug
#   where video frames with SMPTE 432 color primaries are rendered
#   incorrectly when sampled from an external texture in GLES. To work
# ==============================================================================

diff -r 12e266e296c1 -r 290917deea87 gfx/layers/GLImages.h
--- a/gfx/layers/GLImages.h	Wed Dec 13 08:29:00 2023 +0000
+++ b/gfx/layers/GLImages.h	Wed Dec 13 09:06:29 2023 +0000
@@ -54,6 +54,7 @@
   SurfaceTextureImage(AndroidSurfaceTextureHandle aHandle,
                       const gfx::IntSize& aSize, bool aContinuous,
                       gl::OriginPos aOriginPos, bool aHasAlpha,
+                      bool aForceBT709ColorSpace,
                       Maybe<gfx::Matrix4x4> aTransformOverride);
 
   gfx::IntSize GetSize() const override { return mSize; }
@@ -61,6 +62,7 @@
   bool GetContinuous() const { return mContinuous; }
   gl::OriginPos GetOriginPos() const { return mOriginPos; }
   bool GetHasAlpha() const { return mHasAlpha; }
+  bool GetForceBT709ColorSpace() const { return mForceBT709ColorSpace; }
   const Maybe<gfx::Matrix4x4>& GetTransformOverride() const {
     return mTransformOverride;
   }
@@ -102,6 +104,7 @@
   bool mContinuous;
   gl::OriginPos mOriginPos;
   const bool mHasAlpha;
+  const bool mForceBT709ColorSpace;
   const Maybe<gfx::Matrix4x4> mTransformOverride;
   UniquePtr<SetCurrentCallback> mSetCurrentCallback;
 };